### PR TITLE
Development

### PR DESCRIPTION
**Pull request by smolamarcin**

### CODE REVIEW

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
         this.ip = ip;
     }
 
+    @Override
     public boolean run() throws IOException {
```
Comment (korotkevics) [inline battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java]: javadoc

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java [hunk 2]
```diff
@@ -40,7 +41,13 @@ public void send(String allShips) throws IOException {
         enemyShips = in.readLine();
     }
 
+    @Override
     public String getEnemyShips() {
```
Comment (korotkevics) [inline battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java]: javadoc

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java [hunk 3]
```diff
@@ -40,7 +41,13 @@ public void send(String allShips) throws IOException {
...
         return enemyShips;
     }
+
+    @Override
+    public void sendGameOverToOpponent() {
```
Comment (korotkevics) [inline battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java]: javadoc

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketServer.java [hunk 4]
```diff
@@ -48,12 +55,30 @@ public Cell receiveFirstMove() {
         return cells.poll();
     }
 
+    @Override
+    public void sendGameOverToOpponent() {
```
Comment (korotkevics) [inline battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketServer.java]: javadoc

File: battleship-client/src/main/java/com/epam/solid/nie/client/ui/GameScene.java [hunk 5]
```diff
@@ -49,8 +53,16 @@ private Parent createContent() {
                     return;
                 running = cell.shoot();
                 if (checkForWin(enemyBoard)) {
```
Comment (korotkevics) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ui/GameScene.java]: There are so many instructions, consider to extract private methods

File: battleship-client/src/main/java/com/epam/solid/nie/client/ui/GameScene.java [hunk 6]
```diff
@@ -72,15 +84,10 @@ private void makeEnemyMove(Cell cell1) {
             int x = cell1.getCellX();
             int y = cell1.getCellY();
             Cell cell = playerBoard.getCell(x, y);
-            if (cell.wasShot) {
+            if (cell.wasShot)
                 cell1 = socketServer.receiveEnemyMove();
```
Comment (korotkevics) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ui/GameScene.java]: I woulnd't use numbers in variables names, and also adding a number solely for the purpose of distinguishing it from another variable is a bad practice, add cell and cell1 different names

File: battleship-client/src/main/java/com/epam/solid/nie/client/ui/GameScene.java [hunk 7]
```diff
@@ -72,15 +84,10 @@ private void makeEnemyMove(Cell cell1) {
             int x = cell1.getCellX();
             int y = cell1.getCellY();
             Cell cell = playerBoard.getCell(x, y);
-            if (cell.wasShot) {
+            if (cell.wasShot)
```
Comment (korotkevics) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ui/GameScene.java]: braces

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java (hunk 1 above)
Comment (Ozymadias) [reply battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java]: there is separate task for adding javadocs

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java (hunk 2 above)
Comment (Ozymadias) [reply battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java]: there is separate task for adding javadocs

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java (hunk 3 above)
Comment (Ozymadias) [reply battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java]: there is separate task for adding javadocs

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketServer.java (hunk 4 above)
Comment (Ozymadias) [reply battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketServer.java]: same as before

File: battleship-client/src/main/java/com/epam/solid/nie/client/ui/GameScene.java (hunk 7 above)
Comment (Ozymadias) [reply battleship-client/src/main/java/com/epam/solid/nie/client/ui/GameScene.java]: why do you consider lack of braces as mistake - in my opinion it is one of matters which should be determine by each team

Comment (korotkevics) [reply battleship-client/src/main/java/com/epam/solid/nie/client/ui/GameScene.java]: In my opinion (and I've heard/read same many times) they're highly recommended for readability and for lowering the risk of introducing bugs when expanding a simple logic. If you're strongly opposed to it, then you can ignore it..

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java (hunk 1 above)
Comment (korotkevics) [reply battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java]: ok

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java (hunk 2 above)
Comment (korotkevics) [reply battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java]: ok

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketServer.java (hunk 4 above)
Comment (korotkevics) [reply battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketServer.java]: ok

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java (hunk 1 above)
Comment (LIttleAncientForestKami) [reply battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java]: Why are there binary files here? DS_Store ones?

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java (hunk 3 above)
Comment (LIttleAncientForestKami) [reply battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java]: @Ozymadias could you link it please?

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketServer.java [hunk 8]
```diff
@@ -48,12 +55,30 @@ public Cell receiveFirstMove() {
...
+        server.sendGameOverToOpponent();
+    }
+
     private void receiveAllMovesWithoutSending() {
         allMoves = "";
         String moves = server.getEnemyShips();
+        if (moves.equals("Q")) {
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketServer.java]: String equals and capital letter. Intentional? Or did you mean "q" as well?

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketServer.java [hunk 9]
```diff
@@ -48,12 +55,30 @@ public Cell receiveFirstMove() {
...
+            secondStage.setScene(secondScene);
+
+
+            secondStage.show();
+        }
         String[] movesArr = moves.split(",;");
-        for (int i = 0; i < movesArr.length; i++) {
-            String[] coordinates = movesArr[i].split(",");
+        for (String aMovesArr : movesArr) {
+            String[] coordinates = aMovesArr.split(",");
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketServer.java]: :+1:

File: battleship-client/src/main/java/com/epam/solid/nie/client/ui/GameScene.java (hunk 7 above)
Comment (LIttleAncientForestKami) [reply battleship-client/src/main/java/com/epam/solid/nie/client/ui/GameScene.java]: It IS your choice and preference. However, many errors were caused by lack of braces already and adding "just one line" later on. And no - you don't need to change if your team doesn't want to.

File: battleship-client/src/main/java/com/epam/solid/nie/client/ui/StartSceneController.java [hunk 10]
```diff
@@ -25,7 +29,13 @@ void btn_connect_clicked(ActionEvent event) throws IOException {
             boolean whichPlayer = socketServer.connect(ip);
             new GameScene(socketServer, whichPlayer).start();
         } else {
-            System.out.println("Wrong ip.");
+            Label label=new Label("Wrong ip");
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/java/com/epam/solid/nie/client/ui/StartSceneController.java]: IP is an abbreviation, hence, it should be capitalized.

File: battleship-client/src/main/resources/startScreen.fxml [hunk 11]
```diff
@@ -1,5 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
+<?import java.lang.*?>
```
Comment (LIttleAncientForestKami) [inline battleship-client/src/main/resources/startScreen.fxml]: That's the only change? Two imports? How did this happen?

File: battleship-client/src/main/java/com/epam/solid/nie/client/ui/GameScene.java (hunk 7 above)
Comment (Ozymadias) [reply battleship-client/src/main/java/com/epam/solid/nie/client/ui/GameScene.java]: https://github.com/smolamarcin/battleship/issues/151

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketServer.java (hunk 8 above)
Comment (Ozymadias) [reply battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketServer.java]: Intentionally. Is there any problem with it?

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java (hunk 3 above)
Comment (Ozymadias) [reply battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketClient.java]: https://github.com/smolamarcin/battleship/issues/151

File: battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketServer.java (hunk 8 above)
Comment (LIttleAncientForestKami) [reply battleship-client/src/main/java/com/epam/solid/nie/client/communication/SocketServer.java]: Just making sure it's not a typical mistake of "equals vs equalsIgnoreCase".

**Pull request merged**